### DISCUSSION TITLE
Query project support request

### DISCUSSION THREAD

**Discussion by misha-p** [JasperFx/marten#2801] [Ideas]
While trying to migrate to Marten from mongoDB I noticed that some generic projection functionality is missing in Marten which could be quite useful in many cases.
For example with mongoDB I am able to implement the following (using mongoDB ProjectionBuilder)
```cs
interface IMyRepository<T>
{
    T GetById(Guid id, params Expression<Func<T, object>>[] fields);
    T GetById(Guid id, params string[] fields);
}
```

with usage like:
```cs
IMyRepository<MyType> _myRepo;
MyType obj = _myRepo.GetById(
    id, 
    p => p.Id,
    p => p.Property1,
    p => p.Property2,
    p => p.Property3);
```

It would be very helpful if Marten would support this kind of document projecting (e.g. Project extension) which internally would perform partial object initialization without the need to explicitly create original type being queried or use anonymous.

**Comment by jeremydmiller** [comment]
This is to help us use Marten at MedeAnalytics. Creating a shim for some common MongoDb functionality we use heavily at work. This will be a small alternative to using our `Select()` linq support. I'm taking this, and trying to turn it around quickly for 5.9
